django/forms/forms.py

```diff
@@ -580,12 +586,16 @@ def value(self):
         if not self.form.is_bound:
             data = self.form.initial.get(self.name, self.field.initial)
             if callable(data):
-                data = data()
-                # If this is an auto-generated default date, nix the
-                # microseconds for standardized handling. See #22502.
-                if (isinstance(data, (datetime.datetime, datetime.time)) and
-                        not getattr(self.field.widget, 'supports_microseconds', True)):
-                    data = data.replace(microsecond=0)
+                if self._initial_value is not UNSET:
```

Comment:
Ah, I see. I was curious whether that was a common style in Django, etc. Thanks for the reply @timgraham



> On Feb 24, 2015, at 7:49 PM, Tim Graham <notifications@github.com> wrote:
> 
> In django/forms/forms.py:
> 
> > @@ -580,12 +586,16 @@ def value(self):
> >          if not self.form.is_bound:
> >              data = self.form.initial.get(self.name, self.field.initial)
> >              if callable(data):
> > -                data = data()
> > -                # If this is an auto-generated default date, nix the
> > -                # microseconds for standardized handling. See #22502.
> > -                if (isinstance(data, (datetime.datetime, datetime.time)) and
> > -                        not getattr(self.field.widget, 'supports_microseconds', True)):
> > -                    data = data.replace(microsecond=0)
> > +                if self._initial_value is not UNSET:
> I don't really mind either way but discussed briefly in IRC and concluded: "Code is clearer when objects of a given type always have the same set of attributes - the presence of a sentinel value is a clearer marker of intent (as opposed to bug) than the absence of an attribute entirely." -Carl Meyer
> 
> —
> Reply to this email directly or view it on GitHub.
>